src/library/FOSSBilling/Fingerprint.php

```diff
@@ -130,9 +132,34 @@ public function checkFingerprint(array $fingerprint): bool
             }
         }
 
-        // Remove the total score from the total number of items. The final score must be at least half the number of properties in order for the fingerprint to be considered valid.
-        $finalScore = $itemCount - $scoreSubtract;
-        return $finalScore >= ($itemCount / 2);
+        /**
+         * Here we calculate how confident we are in our ability to fingerprint a device without causing issues for legitimate sessions.
+         * The less confident we are, the more wrong the current fingerprint needs to be when compared against the session's before it is invalidated.
+         * 
+         * In the event that less that 70% of the possible values are in the fingerprint, we use the percentage off we are to calculate a higher percentage before failure.
+         * For example:
+         *  If we have 13 possible fingerprint properties and only 9 are available, that's only 69% of the possible properties and the failure threshold will be calculated as follows.
+         *  Negative weight: (1 - 0.69) / 1.25 = `0.24`
+         *  Failure threshold: 0.5 + 0.24 = `0.74`
+         *  
+         *  So in this example, the percentage wrong needs to be at or above 74% (nearly 75%) before the session is declared invalid.
+         *  If there's only 6 of 13 available that moves to 93% and at 5 of 13 it's 99%.
+         *  By doing this, we can effectively give additional headroom in situations where we are less-confident than we'd like to be and effectively completely disable the check in a worst-case situation.
```

Comment:
So if I craft a request to remove the HTTP Headers REMOTE_ADDR, HTTP_REFERER, HTTP_X_FORWARDED_FOR, HTTP_ACCEPT_LANGUAGE, HTTP_ACCEPT_ENCODING, HTTP_UPGRADE_INSECURE_REQUESTS and HTTP_CF_IPCOUNTRY, I've removed 7 of 13 Values, meaning there's only 6 left to compare. Because 93% of the remaining ones need to be false before the session gets invalidated the remaining values are: userAgent, browser, browserVersion, os, HTTP_SEC_CH_UA_PLATFORM and HTTP_SEC_CH_UA_MOBILE. 

So if I get os and browser right, I can still hijack the session, right? 

Windows has 70-75% market share, Macos ~16%. Browsers are 63% Chrome and ~20% Safari.

I think we should find more and different markers, or we need to bring the agressiveness of this additional headroom down.

Reply:
If you removed `REMOTE_ADDR`, `HTTP_REFERER`, `HTTP_X_FORWARDED_FOR`, `HTTP_ACCEPT_LANGUAGE`, `HTTP_ACCEPT_ENCODING`, `HTTP_UPGRADE_INSECURE_REQUESTS`, and `HTTP_CF_IPCOUNTRY` from the header when they were initially there that would instantly be considered to be 100% wrong because the added up weights from those are 13.

Reply:
Any and all properties that are in one and not in the other are counted as wrong.
Any and all properties that do not match are also counted as wrong.

Your example also assumes that their IP address correctly matches the original one when it won't and there is also a fairly decent chance that the browser version will differ, which is also counted as wrong. 

Adding these two changes on top of the example you gave brings the weighted wrong percentage to 120%